Entities/ACrab.cpp

```diff
@@ -2785,11 +2783,7 @@ void ACrab::Update()
     if (m_Status == STABLE)
     {
         // Upright body posture
-        // Break the spring if close to target angle.
-        if (fabs(rot) > (c_HalfPI - c_SixteenthPI))
-            m_AngularVel -= rot * 0.5;//fabs(rot);
-        else if (fabs(m_AngularVel) > 0.3)
-            m_AngularVel *= 0.85;
+		m_AngularVel = m_AngularVel * 0.95F - (rot * 0.2F);
```

Comment:
Dreadnoughts (and all ACrabs) have always rocked back and forth uncontrollably and it is because of the AffectsRotation(?) argument in PushAsLimb (I've given it some testing but haven't reached a conclusion) - it's very unreliable and in dire need of some further looking into. This part of the code is a lazy fix to that problem, but I seem to have forgotten to revert the changes from my testing. I'll re-enable it, I guess.

Reply:
From the bit of testing I did, I found that in your branch crabs tended to never stop rocking back and forth but they do it more slowly and gently, while in pre3 they tend to do be more jittery about it, but sometimes come to a stop. I guess I can't say for sure that it's your changes that did this though, since I used pre3 cause that was the easiest thing for me to test against at the time.

And yeah, this stuff being tied into limb pushing and so on makes it all complex, so I've not dug too deep. In general, the solution to this sort of wiggling is (I think I mentioned it in another comment) to have some deadzone where they just snap to the proper position. As long as the angular velocity and rotation difference for that to happen is small enough, it shouldn't look/feel weird.